master/internal/stream/utils.go

```diff
@@ -0,0 +1,63 @@
+package stream
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"github.com/uptrace/bun"
+
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/stream"
+)
+
+func permFilter(
```

Comment:
nit (non-blocking): Adding some comments here would be helpful in the future so it's clear what the intended use of these functions are